src/parser/core/modules/Combos.tsx

```diff
@@ -20,14 +20,17 @@ const ISSUE_TYPENAMES = {
 	failedcombo: <Trans id="core.combos.issuetypenames.failed">Missed or Invulnerable</Trans>,
 }
 
-export interface ComboEvent extends AoeEvent {
-	type: 'combo'
+export class ComboEvent extends NormalisedDamageEvent {
+	constructor(event: NormalisedDamageEvent) {
+		super(event)
+		this.type = 'combo'
```

Comment:
Hm, being able to do this assignment is a little unfortunate - the base class' `type: string` means filters and hooks will be a little less strict on types.

That said, it's partially an unfortunate side-effect of using classes. There's no _nice_ way of overriding the `type` field's type as a class.

Something to think about for later.

Reply:
In part, that’s why the normalisedEvents module is doing both an interface and a class for the normalisedX classes, which seems to make typescript a bit happier about the event type and keeps it from just being an arbitrary string to TS. But for some reason Combos wouldn’t take that same pattern, and I just kind of gave up on it. Right now there’s only one module that hooks combo events (DRK resource generation), so the damage is fairly contained.